src/renderer/src/lib/State.ts

```diff
@@ -9,7 +9,7 @@ export class State<T extends {}> {
     return this;
   }
 
-  get<K extends keyof T>(key: K): T[K] | undefined {
+  get<K extends keyof T>(key: K): T[keyof T] | undefined {
```

Comment:
This should be `T[K]` because you are getting the value of `K` and you are passing the exact key you want.

Reply:
![image](https://github.com/user-attachments/assets/f194337d-f80a-4ba8-aa77-9a896e3e9133)

Reply:
This should work:

```diff
diff --git a/src/renderer/src/lib/State.ts b/src/renderer/src/lib/State.ts
index 9dd9825..b6ab0e5 100644
--- a/src/renderer/src/lib/State.ts
+++ b/src/renderer/src/lib/State.ts
@@ -1,15 +1,15 @@
-export class State<T extends {}> {
-  private readonly schema: Map<keyof T, T[keyof T]>;
+export class State<T extends {}, K extends keyof T = keyof T> {
+  private readonly schema: Map<K, T[K]>;
   constructor() {
     this.schema = new Map();
   }

-  set<K extends keyof T>(key: K, value: T[K]): State<T> {
+  set(key: K, value: T[K]): State<T> {
     this.schema.set(key, value);
     return this;
   }

-  get<K extends keyof T>(key: K): T[keyof T] | undefined {
+  get(key: K): T[K] | undefined {
     return this.schema.get(key);
   }
 }
```